jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/internal/HttpChannelState.java

```diff
@@ -394,10 +386,10 @@ public Runnable onFailure(Throwable x)
             }
 
             // Set the error to arrange for any subsequent reads, demands or writes to fail.
-            if (_failure == null)
-                _failure = Content.Chunk.from(x, true);
-            else if (ExceptionUtil.areNotAssociated(_failure.getFailure(), x) && _failure.getFailure().getClass() != x.getClass())
-                _failure.getFailure().addSuppressed(x);
+            if (_readFailure == null)
+                _readFailure = Content.Chunk.from(x, true);
+            else if (ExceptionUtil.areNotAssociated(_readFailure.getFailure(), x) && _readFailure.getFailure().getClass() != x.getClass())
+                _readFailure.getFailure().addSuppressed(x);
```

Comment:
Seems like this else branch could just be ...

```suggestion
            else
                ExceptionUtil.addSuppressedIfNotAssociated(_readFailure.getFailure(), x);
```

Reply:
The logic is different as there is the check on the class.